test/metrics/metrics_test.go

```diff
@@ -0,0 +1,311 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/codeready-toolchain/toolchain-common/pkg/states"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	authsupport "github.com/codeready-toolchain/toolchain-common/pkg/test/auth"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/md5"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestMetricsOnWhenUsersReactivated activates and deactivates a few users, and check the metrics.
+// user-0001 will be activated 1 time
+// user-0002 will be activated 2 times
+// user-0003 will be activated 3 times
+func TestMetricsOnWhenUsersDeactivatedAndReactivated(t *testing.T) {
+	// given
+	hostAwait, memberAwait, _ := WaitForDeployments(t)
+	hostAwait.UpdateToolchainConfig(testconfig.AutomaticApproval().Disabled())
+	// host metrics should be available at this point
+	VerifyHostMetricsService(t, hostAwait)
+	VerifyMemberMetricsService(t, memberAwait)
+	metricsAssertion := InitMetricsAssertion(t, hostAwait, []string{memberAwait.ClusterName})
+	usersignups := map[string]*toolchainv1alpha1.UserSignup{}
+
+	// when
+	for i := 1; i <= 3; i++ {
+		username := fmt.Sprintf("user-%04d", i)
+		usersignups[username] = CreateAndApproveSignup(t, hostAwait, username, memberAwait.ClusterName)
+		for j := 1; j < i; j++ { // deactivate and reactivate as many times as necessary (based on its "number")
+			// deactivate the user
+			_, err := hostAwait.UpdateUserSignupSpec(usersignups[username].Name, func(usersignup *toolchainv1alpha1.UserSignup) {
+				states.SetDeactivated(usersignup, true)
+			})
+			require.NoError(t, err)
+			err = hostAwait.WaitUntilMasterUserRecordDeleted(username)
+			require.NoError(t, err)
+			// reactivate the user
+			CreateAndApproveSignup(t, hostAwait, username, memberAwait.ClusterName, WithIdentityID(usersignups[username].Spec.Userid))
+		}
+	}
+
+	// then verify the value of the `sandbox_users_per_activations` metric
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "1", "domain", "external") // 1 activation
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "2", "domain", "external") // 1 activation
+	metricsAssertion.WaitForMetricDelta(UsersPerActivationsAndDomainMetric, 1, "activations", "3", "domain", "external") // 1 activation
```

Comment:
you could also verify the deactivation metric here. And probably also the others like `UserSignupsMetric` `UserSignupsApprovedMetric` ...